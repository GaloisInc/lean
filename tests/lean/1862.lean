variable R : Type
variable [ring R]

example : -(-(1:R)) = 1 :=
begin
  trace_state,
  exact neg_neg 1,
end

#check - -(1:R)
