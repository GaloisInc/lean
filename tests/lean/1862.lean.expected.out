R : Type,
_inst_1 : ring R
‚ä¢ - -1 = 1
- -1 : R
